### PR TITLE
(SIMP-6270) Remove bracketize 

### PR DESCRIPTION
**Pull request by sharkbruhaha**

### CODE REVIEW

**Comment by sharkbruhaha** [comment]
I am not sure what was originally going on in the module, but bracketize isn't needed in this case. It is currently being used on the listener port, which is always just an Integer. The bracketize function as written was doing nothing at all to the value that was passed to it because it was not an IPv6 address [with netmask]. 

I guess if you wanted a really fancy flattening... but that can just be done without the function.

**Comment by sharkbruhaha** [comment]
Bracketize for reference:
`
module Puppet::Parser::Functions
  newfunction(:bracketize, :type => :rvalue, :doc => <<-'ENDHEREDOC') do |ipaddr|
    Add brackets to IP addresses and `Arrays` of IP addresses based on the
    rules for bracketing IPv6 addresses.

    Ignore anything that does not look like an IPv6 address.

    @return [Variant[String, Array[String]]]
    ENDHEREDOC

    ipaddr = Array(ipaddr).flatten
    toret = ipaddr.map { |x|
      # IPv6 Address?
      if x.include?(':') and x !~ /\./ then
        if x[0].chr != '[' then
          y = x.split('/')
          y[0] = "[#{y[0]}]"
          y.join('/')
        end
      else
        x
      end
    }

    toret = toret.first if toret.size == 1
    return toret
  end
end
`

**Comment by trevor-vaughan** [comment]
> 
> 
> I am not sure what was originally going on in the module, but bracketize isn't needed in this case. It is currently being used on the listener port, which is always just an Integer. The bracketize function as written was doing nothing at all to the value that was passed to it because it was not an IPv6 address [with netmask].
> 
> I guess if you wanted a really fancy flattening... but that can just be done without the function.

You have definitely identified a bug. All  `listen` parameters should be `Array[Variant[Simplib::Host::Port, Simplib::Port]` which is why the `bracketize` function is necessary.

**Comment by sharkbruhaha** [comment]
> @sharkbruhaha See comments for the bug that you uncovered.

Will handle in bug ticket, making changes for bracketizing and closing this one.

bug ticket: https://simp-project.atlassian.net/browse/SIMP-6324

**Comment by sharkbruhaha** [comment]
As per our discussion @trevor-vaughan I've just rolled the type update into this ticket and am keeping the bug ticket around in backlog.

File: templates/etc/httpd/conf.d/ssl.conf.erb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 </IFModule>
 
 <% @listen.each do |l| -%>
-Listen <%= scope.function_bracketize([l.to_s]) %><%= "\n" %>
+Listen <%= scope.call_function('simp::bracketize', [l.to_s]) %><%= "\n" %>
```
Comment (trevor-vaughan) [inline templates/etc/httpd/conf.d/ssl.conf.erb]:
```suggestion
Listen <%= scope.call_function('simplib::bracketize', [l.to_s]) %><%= "\n" %>
```

File: templates/etc/httpd/conf/httpd.conf.erb [hunk 2]
```diff
@@ -57,7 +57,7 @@ MaxRequestsPerChild  <%= @worker_maxrequestsperchild %>
 
 # Default: 80
 <% @listen.flatten.each do |l| -%>
-Listen <%= scope.function_bracketize([l.to_s]) %>
+Listen <%= scope.call_function('simp::bracketize', [l.to_s]) %>
```
Comment (trevor-vaughan) [inline templates/etc/httpd/conf/httpd.conf.erb]:
```suggestion
Listen <%= scope.call_function('simplib::bracketize', [l.to_s]) %>
```

File: CHANGELOG [hunk 3]
```diff
@@ -1,3 +1,9 @@
+* Fri Mar 22 2019 Joseph Sharkey <shark.bruhaha@gmail.com> - 6.1.2-0
```
Comment (trevor-vaughan) [inline CHANGELOG]: Please collapse these into one changelog entry

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-simp_apache/pipelines/53505010

**Pull request merged**